projects/Reports/template-sql/population-counts.template.sql

```diff
@@ -0,0 +1,280 @@
+HAVING MIN(CodingOptOutFlag) = MAX(CodingOptOutFlag);
+
+-- Tidy up - helpful in ensuring the tempdb doesn't run out of space mid-query
+DROP TABLE #UnmatchedPatientOptOut;
+
+
+-- Find patients who died (ID, Deceased)=====================================================================================================================================
+IF OBJECT_ID('tempdb..#PatientsDied') IS NOT NULL DROP TABLE #PatientsDied;
+SELECT PK_Patient_Link_ID, Deceased
+INTO #PatientsDied
+FROM [RLS].[vw_Patient_Link];
+
+
+-- Find patients who have GP event records (ID, ID_GP_Event)==========================================================================================================================
+IF OBJECT_ID('tempdb..#PatientsGPEvent') IS NOT NULL DROP TABLE #PatientsGPEvent;
+SELECT DISTINCT FK_Patient_Link_ID AS ID_GP_Event
+INTO #PatientsGPEvent
+FROM RLS.vw_GP_Events;
+
+
+-- Find patients who have GP medication records (ID, ID_GP_Medication)==========================================================================================================================
+IF OBJECT_ID('tempdb..#PatientsGPMedication') IS NOT NULL DROP TABLE #PatientsGPMedication;
+SELECT DISTINCT FK_Patient_Link_ID AS ID_GP_Medication
+INTO #PatientsGPMedication
+FROM RLS.vw_GP_Medications;
+
+
+-- Merge all information into 1 table============================================================================================================================
+-- Merge into 1 table
+IF OBJECT_ID('tempdb..#Population') IS NOT NULL DROP TABLE #Population;
+SELECT p.FK_Patient_Link_ID, gp.Tenancy_ID, lsoa.LSOA_Code, lsoa.GM_Area, gpgm.GP_Code, gpgm.CCG, opt.CodingOptOutFlag, die.Deceased, e.ID_GP_Event, m.ID_GP_Medication 
+INTO #Population
+FROM #Patients p
+LEFT OUTER JOIN #PatientsWithGP gp ON gp.FK_Patient_Link_ID = p.FK_Patient_Link_ID
+LEFT OUTER JOIN #PatientsLSOAGM lsoa ON lsoa.FK_Patient_Link_ID = p.FK_Patient_Link_ID
+LEFT OUTER JOIN #PatientsGPGM gpgm ON gpgm.FK_Patient_Link_ID = p.FK_Patient_Link_ID
+LEFT OUTER JOIN #PatientsOptOut opt ON opt.FK_Patient_Link_ID = p.FK_Patient_Link_ID
+LEFT OUTER JOIN #PatientsDied die ON die.PK_Patient_Link_ID = p.FK_Patient_Link_ID
+LEFT OUTER JOIN #PatientsGPEvent e ON e.ID_GP_Event = p.FK_Patient_Link_ID
+LEFT OUTER JOIN #PatientsGPMedication m ON m.ID_GP_Medication = p.FK_Patient_Link_ID;
+
+SELECT TOP (50) *
+FROM #Population;
+
+-- Tidy up - helpful in ensuring the tempdb doesn't run out of space mid-query
+DROP TABLE #Patients;
+DROP TABLE #PatientsWithGP;
+DROP TABLE #PatientsLSOAGM;
+DROP TABLE #PatientsGPGM;
+DROP TABLE #PatientsOptOut;
+DROP TABLE #PatientsDied;
+DROP TABLE #PatientsGPEvent;
+DROP TABLE #PatientsGPMedication;
+
+
+-- Get information for GMCR population===========================================================================================================================
+-- Today date
+SELECT FORMAT(GETDATE(), 'yyyy-MM-dd');
+
+-- All patients in GMCR
+SELECT COUNT (*) FROM #Population;
```

Comment:
For all of these `SELECT COUNT(*)` statements you should add a label e.g. `SELECT COUNT(*) AS TotalPopulation`. It just means that when you run them all at once you can see at a glance in the results which each number refers to.

Alterntively you could do something like this to bring them all into a single `SELECT` statement:

```sql
SELECT 
  COUNT(*) AS TotalPopulation,
  SUM(CASE WHEN Tenancy_ID = 2 THEN 1 ELSE 0 END) AS GPRegistered,
  SUM(CASE WHEN Tenancy_ID IS NULL THEN 1 ELSE 0 END) AS NotGPRegistered,
  -- ... more lines here...
FROM #Population
```